packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -11,24 +11,1644 @@ import { PerformanceConfigurator } from "../Engines/performanceConfigurator";
+     */
+    static Hermite1stDerivative<T extends Vector>(value1: DeepImmutable<T>, tangent1: DeepImmutable<T>, value2: DeepImmutable<T>, tangent2: DeepImmutable<T>, time: number): T;
+
+    /**
+     * Returns a new Vector which is the 1st derivative of the Hermite spline defined by the vectors "value1", "value2", "tangent1", "tangent2".
+     * @param value1 defines the first control point
+     * @param tangent1 defines the first tangent
+     * @param value2 defines the second control point
+     * @param tangent2 defines the second tangent
+     * @param time define where the derivative must be done
+     * @param result define where the derivative will be stored
+     * @returns result input
+     */
+    static Hermite1stDerivativeToRef<T extends Vector>(
+        value1: DeepImmutable<T>,
+        tangent1: DeepImmutable<T>,
+        value2: DeepImmutable<T>,
+        tangent2: DeepImmutable<T>,
+        time: number,
+        result: T
+    ): T;
+
+    /**
+     * Returns a new Vector located for "amount" (float) on the linear interpolation between the vector "start" adn the vector "end".
+     * @param start defines the start vector
+     * @param end defines the end vector
+     * @param amount defines the interpolation factor
+     * @returns a new Vector
+     */
+    static Lerp<T extends Vector>(start: DeepImmutable<T>, end: DeepImmutable<T>, amount: number): T;
+
+    /**
+     * Returns a new Vector located for "amount" (float) on the linear interpolation between the vector "start" adn the vector "end".
+     * @param start defines the start vector
+     * @param end defines the end vector
+     * @param amount defines the interpolation factor
+     * @returns a new Vector
+     */
+    static LerpToRef<T extends Vector>(start: DeepImmutable<T>, end: DeepImmutable<T>, amount: number, result: T): T;
+
+    /**
+     * Returns a new Vector equal to the normalized given vector
+     * @param vector defines the vector to normalize
+     * @returns a new Vector
+     */
+    static Normalize<T extends Vector>(vector: DeepImmutable<T>): T;
+
+    /**
+     * Normalize a given vector into a second one
+     * @param vector defines the vector to normalize
+     * @param result defines the vector where to store the result
+     * @returns result input
+     */
+    static NormalizeToRef<T extends Vector>(vector: DeepImmutable<T>, result: T): T;
+
+    /**
+     * Gets the shortest distance (float) between the point "p" and the segment defined by the two points "segA" and "segB".
+     * @param p defines the middle point
+     * @param segA defines one point of the segment
+     * @param segB defines the other point of the segment
+     * @returns the shortest distance
+     */
+    static DistanceOfPointFromSegment<T extends Vector>(p: DeepImmutable<T>, segA: DeepImmutable<T>, segB: DeepImmutable<T>): number;
+}
+
+/**
+ * Represents a vector in any dimensional space
+ */
+export class DynamicVector implements Vector<number[]> {

```

Comment:
This should not be here. Nothing is actually using it in the framework but it would be included in every growing the size for no reasons. I am not opposed to have it but it needs to be in its own module. I would also like to see a use for it :-)

Reply:
Removed in [157fc18](https://github.com/BabylonJS/Babylon.js/pull/13699/commits/157fc18cd530c9949d920475cb52ee5670b09539)